lib/spack/spack/cmd/list.py

```diff
@@ -40,18 +50,27 @@ def setup_parser(subparser):
         help='Optional glob patterns to filter results.')
     subparser.add_argument(
         '-s', '--sensitive', action='store_true', default=False,
-        help='Use case-sensitive filtering. Default is case sensitive, '
+        help='Use case-sensitive filtering. Default is case insensitive, '
         'unless the query contains a capital letter.')
```

Comment:
I just tested this out and apparently it is actually case sensitive. Try running:
```
$ spack list imagemagick
```
Honestly, there are so few packages in Spack with capital letters that I think we should always be case insensitive. We might just want to remove the option altogether.